internal/provider/resource_tfe_team_notification_configuration.go

```diff
@@ -0,0 +1,437 @@
+				Description: "The HTTP or HTTPS URL where notification requests will be made. This value must not be provided if `destination_type` is `email`.",
+				Optional:    true,
+				Validators: []validator.String{
+					validators.AttributeRequiredIfValueString(
+						"destination_type",
+						[]string{"generic", "microsoft-teams", "slack"},
+					),
+					validators.AttributeValueConflictValidator(
+						"destination_type",
+						[]string{"email"},
+					),
+					stringvalidator.ConflictsWith(
+						path.MatchRelative().AtParent().AtName("email_addresses"),
+						path.MatchRelative().AtParent().AtName("email_user_ids"),
+					),
+				},
+			},
+
+			"team_id": schema.StringAttribute{
+				Description: "The ID of the team that owns the notification configuration.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+	}
+}
+
+// Configure implements resource.ResourceWithConfigure
+func (r *resourceTFETeamNotificationConfiguration) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(ConfiguredClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected resource Configure type",
+			fmt.Sprintf("Expected tfe.ConfiguredClient, got %T. This is a bug in the tfe provider, so please report it on GitHub.", req.ProviderData),
+		)
+	}
+	r.config = client
+}
+
+func (r *resourceTFETeamNotificationConfiguration) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan modelTFETeamNotificationConfiguration
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get team
+	teamID := plan.TeamID.ValueString()
+
+	// Create a new options struct
+	options := tfe.NotificationConfigurationCreateOptions{
+		DestinationType: tfe.NotificationDestination(tfe.NotificationDestinationType(plan.DestinationType.ValueString())),
+		Enabled:         plan.Enabled.ValueBoolPointer(),
+		Name:            plan.Name.ValueStringPointer(),
+		Token:           plan.Token.ValueStringPointer(),
+		URL:             plan.URL.ValueStringPointer(),
+		SubscribableChoice: &tfe.NotificationConfigurationSubscribableChoice{
+			Team: &tfe.Team{ID: teamID},
+		},
+	}
+
+	// Add triggers set to the options struct
+	var triggers []types.String
+	if diags := plan.Triggers.ElementsAs(ctx, &triggers, true); diags != nil && diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	options.Triggers = []tfe.NotificationTriggerType{}
+	for _, trigger := range triggers {
+		options.Triggers = append(options.Triggers, tfe.NotificationTriggerType(trigger.ValueString()))
+	}
+
+	// Add email_addresses set to the options struct
+	emailAddresses := make([]types.String, 0)
```

Comment:
Consider:

```suggestion
	emailAddresses := make([]types.String, len(plan.Triggers.Elements()))
```

There are some other places in the PR where we can allocate to a known length.

Reply:
these should all be good now